### PR TITLE
Add software versions in nf-test and check column presence in `.Rmd`

### PR DESCRIPTION
**Pull request by atrigila**
This PR addresses the following:

- [x] Check for the gene biotype column before plotting in the `.Rmd` . If the column is missing, plotting is skipped.
- [x] Remove `--changed-since HEAD^` from `ci.yml` to ensure `.Rmd` file changes are detected, triggering test re-runs. I noticed that since we moved to nf-tests in other PRs these tests are not running due to this line. 
- [x] Update `nf.tests` to use `removeNextflowVersion("$outputDir/pipeline_info/collated_versions.yml")`, now verifying all software versions used in the pipeline.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [x] `CHANGELOG.md` is updated.

### CODE REVIEW

File: assets/differentialabundance_report.Rmd [hunk 1]
```diff
@@ -896,24 +896,28 @@ for (i in 1:nrow(contrasts)){
...
-            labs(
-                title = paste0("Differentially Expressed Genes by Gene Biotype (", dir, ")"),
-                x = "Gene Biotype",
-                y = "Number of Differentially Expressed Genes"  ) +
-            theme_minimal() +
-            theme(axis.text.x = element_text(angle = 45, hjust = 1))
-
-        print(gene_biotype_plot)
-
+        # Check if Gene biotype column exists
```
Comment (pinin4fjords) [inline assets/differentialabundance_report.Rmd]: No all your doing, but could you see if you can make the indenting less mad before we merge this?

**Review by pinin4fjords — CHANGES_REQUESTED** [review]
Really nice work, thank you, one tiny thing

File: assets/differentialabundance_report.Rmd (hunk 1 above)
Comment (atrigila) [reply assets/differentialabundance_report.Rmd]: Thank you for your feedback, now it should be easier to see the changes :)

File: assets/differentialabundance_report.Rmd [hunk 2]
```diff
@@ -896,6 +896,7 @@ for (i in 1:nrow(contrasts)){
             contrast_de <- round_dataframe_columns(contrast_de, digits=params$report_round_digits)
             print( htmltools::tagList(datatable(contrast_de, caption = paste('Differential genes', dir, 'in', contrast_descriptions[i], " (check", differential_files[[i]], "for more detail)"), rownames = FALSE) ))
 
+        if ("Gene biotype" %in% colnames(contrast_de)) {
```
Comment (pinin4fjords) [inline assets/differentialabundance_report.Rmd:899]: Sorry, didn't mean for you to completely remove indenting, it just didn't seem consistent when I looked yesterday (apologies if it was just me having a bad day) - can you indent the lines inside?

I'll approve in the meantime so you're not blocked when that's done.

**Review by pinin4fjords — APPROVED** [review]

**Pull request merged**